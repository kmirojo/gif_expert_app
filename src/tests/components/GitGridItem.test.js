import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("<GifGridItem />", () => {

    const id = "1";
    const title = "Un titulo";
    const url = "https://localhost/algo.jpg";

    const wrapper = shallow(
        <GifGridItem
            title={title}
            url={url}
        />
    );

    test("Should match Snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a p tag with title', () => {
        const p = wrapper.find("p");
        
        expect(p.text().trim()).toBe( title );
    });

    test('should have image same as props url and alt as title', () => {
        const img = wrapper.find("img");
        
        expect(img.props().src).toBe( url );
        expect(img.props().alt).toBe( title );
    });
    
    test('should have animate__fadeIn Class', () => {
        const div = wrapper.find(".animate__fadeIn");

        expect(div).toHaveLength(1);
    });
});