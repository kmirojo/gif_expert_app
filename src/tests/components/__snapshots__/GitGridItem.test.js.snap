// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> Should match Snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
